databricks/koalas/frame.py

```
@@ -4908,6 +4909,155 @@ def drop_duplicates(self, subset=None, inplace=False):
+        >>> df = ks.DataFrame({'A': [1, 2, 2, 3], 'B': [4, 3, 2, 1]}, columns= ['A', 'B'])
+        >>> df
+           A  B
+        0  1  4
+        1  2  3
+        2  2  2
+        3  3  1
+
+        >>> df.rank().sort_index()
+             A    B
+        0  1.0  4.0
+        1  2.5  3.0
+        2  2.5  2.0
+        3  4.0  1.0
+
+        >>> df.rank(method='min').sort_index()
+             A    B
+        0  1.0  4.0
+        1  2.0  3.0
+        2  2.0  2.0
+        3  4.0  1.0
+
+        >>> df.rank(method='dense').sort_index()
+             A    B
+        0  1.0  4.0
+        1  2.0  3.0
+        2  2.0  2.0
+        3  3.0  1.0
+
+        >>> df = ks.DataFrame({'A': [1, 2, None, 3], 'B': [4, None, 2, 1]}, columns= ['A', 'B'])
+        >>> df
+             A    B
+        0  1.0  4.0
+        1  2.0  NaN
+        2  NaN  2.0
+        3  3.0  1.0
+
+        >>> df.rank().sort_index()
+             A    B
+        0  1.0  3.0
+        1  2.0  NaN
+        2  NaN  2.0
+        3  3.0  1.0
+
+        >>> df.rank(na_option='bottom').sort_index()
+             A    B
+        0  1.0  3.0
+        1  2.0  4.0
+        2  4.0  2.0
+        3  3.0  1.0
+        """
+        if method not in ['average', 'min', 'max', 'first', 'dense']:
+            msg = "method must be one of 'average', 'min', 'max', 'first', 'dense'"
+            raise ValueError(msg)
+
+        if na_option not in ['keep', 'top', 'bottom']:
+            msg = "na_option must be one of 'keep', 'top', 'bottom'"
+            raise ValueError(msg)
+
+        if ascending:
+            asc_func = spark.functions.asc
+        else:
+            asc_func = spark.functions.desc
+
+        index_column = self._internal.index_columns[0]
+        data_columns = self._internal.data_columns
+        sdf = self._sdf
+
+        for column_name in data_columns:
+            if na_option == 'top':
+                if ascending:
+                    fill_value = sdf.agg({column_name: 'min'}).collect()[0][0] - 1
```

Let's avoid this way .. it will be pretty expensive and doesn't lazily execute them.. `.collect()` will actually trigger the computation.

Seems difficult to implement for now. We can simply throw an error that says we don't support it for now.